tools/PI/DevHome.PI/ViewModels/WERPageViewModel.cs

```diff
@@ -69,7 +78,8 @@ public WERPageViewModel()
         AttachedToApp = attachedApp is not null;
         LocalCollectionEnabledForApp = attachedApp is not null ? _werHelper.IsCollectionEnabledForApp(attachedApp + ".exe") : false;
 
-        ((INotifyCollectionChanged)_werHelper.WERReports).CollectionChanged += WER_CollectionChanged;
+        _werAnalyzer = Application.Current.GetService<WERAnalyzer>();
```

Comment:
`WERPageViewModel` seem to be registered to and resolved from the container. Consider consuming `WERAnalyzer` from the constructor argument so it's injected at runtime.

Reply:
Can you elaborate? Are you saying the WERPageViewModel constructor should take a WERAnalyzer vs acquiring it this way?

Reply:
Amir stopped by... I had misunderstood the conversation above. There is a bit of sorcery involved here, but I understand it now.